File: javascript/1-50/2.js

function evenFib(ceiling){
  let curr = 0,
    next = 1,
    sum = 0,
    result = 0;

  while (curr + next < 4000000) {
    sum = curr + next;
    curr = next;
    next = sum;

    if (sum % 2 === 0) {
      result += sum;
    }
  }
  console.log(result);
}

evenFib(4000000);

File: go/solutions/2.go
package solutions 

func FibSeq() int {
	result := 0
	current := 1
	next := 2
	
	for current <= 4000000 {
		if current % 2 == 0 {
			result += current
		}
		current, next = next, current + next
	}
	return result
}